src/main/java/se/simonsoft/cms/transform/service/TransformServiceXsl.java

```diff
@@ -0,0 +1,289 @@
+				}
+				patchset.addLock(locks.getSingle());
+				FileModificationLocked fileMod = new FileModificationLocked(relPath, transformStream);
+				fileMod.setPropertyChange(properties);
+				patchset.add(fileMod);
+			} else {
+				throw new IllegalStateException("Item already exists, config prohibiting overwrite of existing items.");
+			}
+			
+		} catch (IOException e) {
+			throw new RuntimeException("Failed to read stream from transform.", e);
+		} catch (EmptyStreamException e) {
+			logger.warn("Transform of item at path: '{}'  resulted in empty document, will be discarded.", relPath);
+		}
+	}
+	
+	private CmsItemPath getOutputPath(CmsItemId itemId, String output) {
+		
+		CmsItemPath pathResult;
+		
+		if (output == null || output.trim().isEmpty()) {
+			pathResult = new CmsItemIdArg(itemId.getRepository(), itemId.getRelPath().getParent()).getRelPath();
+			logger.debug("Output folder is not specified will default to items parent folder.");
+		} else {
+			pathResult = new CmsItemIdArg(itemId.getRepository(), new CmsItemPath(output)).getRelPath();
+			logger.debug("Output folder is specified: {}", output);
+		}
+		
+		return pathResult;
+	}
+
+	private TransformerService getTransformerService(CmsItemId itemId, String stylesheet) {
+		
+		TransformerService resultService;
+		
+		if (stylesheet.startsWith("/")) {
+			CmsItemId styleSheetItemId = new CmsItemIdArg(itemId.getRepository(), new CmsItemPath(stylesheet));
+			logger.debug("Using stylesheet from CMS: {}", styleSheetItemId.getLogicalId());
+			CmsItem styleSheetItem = itemLookup.getItem(styleSheetItemId);
+			
+			ByteArrayOutputStream baos = new ByteArrayOutputStream();
+			styleSheetItem.getContents(baos);
+			resultService = transformerServiceFactory.buildTransformerService(new StreamSource(IOUtils.toInputStream(baos.toString(StandardCharsets.UTF_8), StandardCharsets.UTF_8)));
+		} else {
+			logger.debug("Using CMS built in stylesheet: {}", stylesheet);
+			resultService = stylesheets.get(stylesheet); 
+		}
+		
+		return resultService;
+	}
+	
+	private boolean pathExists(CmsRepository repo, CmsItemPath path) {
+		
+		CmsItemId itemIdOutput = new CmsItemIdArg(repo, path);
+		boolean result = false;
+		
+		try {
+			itemLookup.getItem(itemIdOutput);
+			result = true;
+		} catch (CmsItemNotFoundException e) {
+			result = false;
+		}
+		
+		return result;
+	}
+	
+	private InputStream checkStreamIsNotEmpty(InputStream inputStream) throws IOException, EmptyStreamException {
+		PushbackInputStream pushbackInputStream = new PushbackInputStream(inputStream);
+		int b;
+		b = pushbackInputStream.read();
+		if ( b == -1 ) {
+			throw new EmptyStreamException("Transform is empty");
+		}
+		pushbackInputStream.unread(b);
+		return pushbackInputStream;
+	}
+	
+	private CmsItemPropertiesMap getProperties(CmsItem baseItem, TransformConfig config) {
+		
+		CmsItemPropertiesMap m = new CmsItemPropertiesMap();
+		m.and(TRANSFORM_BASE_PROP_KEY, baseItem.getId().getLogicalId())
```

Comment:
.and makes sense when chaining and using its return. This just happens to work because it is not immutable.

When used this way, I think multiple put statements make most sense.

Reply:
Fixed.